core.js

```diff
@@ -144,6 +144,18 @@ class FileTypeParser {
 			};
 		}
 
+		if (
+			this.check([0x78, 0x01])
+			|| this.check([0x78, 0x5E])
+			|| this.check([0x78, 0x9C])
+			|| this.check([0x78, 0xDA])
+		) {
```

Comment:
The magic signatures represent:
```
 CMF |  FLG
0x78 | 0x01 - No compression/low
0x78 | 0x5E - Fastest compression
0x78 | 0x9C - Default compression
0x78 | 0xDA - Best compression 
```
The assumptions here is:
- CM (Compression method) = 8 = Deflate (this assumption seems solid)
- CINFO (Compression info) = 7 = 32K window size
- FDICT (preset dictionary) = 0 = not present

I wonder if we could cover corner cases utilizing:
> The FCHECK value must be such that CMF and FLG, when viewed as
>          a 16-bit unsigned integer stored in MSB order (CMF*256 + FLG),
>          is a multiple of 31.

Ref: https://www.rfc-editor.org/rfc/rfc1950#page-5